packages/smooth_app/lib/cache/files/files_cache_impl.dart

```diff
@@ -0,0 +1,121 @@
+// ignore_for_file: avoid_classes_with_only_static_members
+// ignore_for_file: avoid_slow_async_io
+part of 'files_cache.dart';
+
+/// Implementation based on [getTemporaryDirectory]
+class _ShortLivingFilesCacheManagerImpl extends _FilesCacheManagerImpl {
+  _ShortLivingFilesCacheManagerImpl._() : super._();
+
```

Comment:
Just a suggestion:
* put an abstract method `Future<Directory> getDirectory()` in `_FilesCacheManagerImp`
* implement it in both children
* and move `_initCache` one level up to `_FilesCacheManagerImp`

Reply:
`_initCache` is already in `_FilesCacheManagerImp `
Where would to move it exactly?